cirq-core/cirq/ops/measurement_gate.py

```diff
@@ -69,6 +69,16 @@ def __init__(
         if self.invert_mask is not None and len(self.invert_mask) > self.num_qubits():
             raise ValueError('len(invert_mask) > num_qubits')
 
+    @property
+    def key(self) -> str:
+        return str(self.mkey)
+
+    @key.setter
+    def key(self, key_str: str):
+        # TODO: Allow nested keys only when created via some special methods. Planned in Phase 2a
+        #   of https://tinyurl.com/structured-measurement-keys#heading=h.zafcj653k11m
+        self.mkey = value.MeasurementKey(key_str, allow_nested_key=True)
```

Comment:
would it make more sense to set `allow_nested_key` to `False` until the full logic is in place? Otherwise we could break stuff in between.

Reply:
Actually, I'm setting it to `True` to ensure nothing breaks. Currently, a decomposed `CircuitOperation` will has these nested keys inside regular `MeasurementGate`s (nested keys are just keys with a `:` inside). 

Until [Phase 2a of the RFC](https://tinyurl.com/structured-measurement-keys#heading=h.zafcj653k11m), the `MeasurementKey` and `MeasurementGate` classes will not know if their keys are being remapped in a way that they need to allow nested keys. So setting a blanket allow to ensure nothing triggers the `ValueError` (except if someone creates a nested `MeasurementKey` directly).